src/shared/hooks/contracts/collective-rewards/useReadGauges.ts

```diff
@@ -0,0 +1,65 @@
+import { getAbi, type GaugeAbi } from '@/lib/abis/v2'
+import { AVERAGE_BLOCKTIME } from '@/lib/constants'
+import { useMemo } from 'react'
+import { Abi } from 'viem'
+import { UseReadContractParameters, UseReadContractReturnType, useReadContracts } from 'wagmi'
+import { UseReadContractsConfig, ViewPureFunctionName } from '../types'
+
+type GaugeFunctionName = ViewPureFunctionName<GaugeAbi>
+
+export type ReadGaugesConfig<TFunctionName extends GaugeFunctionName> = UseReadContractsConfig<
+  GaugeAbi,
+  TFunctionName
+>
+
+type ReadGaugesReturnType<
+  TAbi extends Abi,
+  TFunctionName extends ViewPureFunctionName<TAbi>,
+> = UseReadContractReturnType<TAbi, TFunctionName>['data']
+
+const abi = getAbi('GaugeAbi')
+
+export const useReadGauges = <TFunctionName extends GaugeFunctionName>(
+  { addresses, ...config }: ReadGaugesConfig<TFunctionName>,
+  query?: Omit<UseReadContractParameters<GaugeAbi, TFunctionName>['query'], 'select'>,
+) => {
+  const { data: results, ...queryData } = useReadContracts({
+    contracts: addresses.map(address => ({
+      abi,
+      address,
+      ...(config as any),
+    })),
+    query: {
+      retry: true,
+      refetchInterval: AVERAGE_BLOCKTIME,
+      ...query,
+    },
+  })
+
+  type ReturnType = ReadGaugesReturnType<GaugeAbi, TFunctionName>
+
+  const data = useMemo(
+    () =>
+      results?.reduce<ReturnType[]>((acc, { result, error, status }, i) => {
```

Comment:
What's the point of iterating over the results if we still return all of them? I think we could have an additional parameter somewhere to throw an error if any of them fails or ignore any error, but very likely it's better for the user to take care of this

Reply:
We decided that this will be handled in separate task